### PR TITLE
Remove deprecated usages and update gems

### PR DESCRIPTION
**Pull request by sai-bhargav**
Updating the gem to pass arguments as keywords in  `lib/routemaster/api_client.rb:76` - to fix spec failures in https://deliveroo.atlassian.net/browse/AVY-2051

Changes made to fix the build:

- Removing `require 'faraday/request/retry'` from `lib/routemaster/api_client.rb`. This is not mentioned to be required as per the docs (Its only require if version >=2 ). Previously faraday version pointed to (1.8.0) now its (1.10.3),  in this version the retry module was moved into another gem `faraday-retry`. (PR - https://github.com/lostisland/faraday/pull/1367)

- CircleCI builds fail for ruby 2.4.1, since orderweb is already on 2.7.7, updated the build to the closest available image(2.7.4)

- Setting sidekiq gem version(version is taken from last successful gem published build), since latest version points to sidekiq 7, which is causing spec failures.

### CODE REVIEW

**Review by jamie-lynch — COMMENTED** [review]
LGTM. Can you also update the CHANGELOG? 

Looking at previous PRs, changes have been added into the `HEAD` section and then a follow up PR has bumped the version and edited the changelog to specify the new version which those changes are included in.

**Review by jamie-lynch — APPROVED** [review]

**Review by prachi-deliveroo — APPROVED** [review]
LGTM

**Review by vanditsmehta — APPROVED** [review]
lgtm

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ version: 2
 jobs:
   build_2.5_rails_5:
     docker:
-      - image: circleci/ruby:2.4.1
+      - image: circleci/ruby:2.7.4
```
Comment (rfwroo) [inline .circleci/config.yml:8]: Worth adding a Ruby 3 build step?

File: gemfiles/rails_3.gemfile [hunk 2]
```diff
@@ -11,7 +11,7 @@ gem "dotenv", :require => false
 gem "simplecov", :require => false
 gem "codecov", :require => false
 gem "resque", :require => false
-gem "sidekiq", :require => false
+gem "sidekiq", '~> 5.2.10', :require => false
```
Comment (rfwroo) [inline gemfiles/rails_3.gemfile]: Sidekiq 5 and Rails 3 [aren't compatible](https://github.com/mperham/sidekiq/blob/main/docs/5.0-Upgrade.md#whats-new), not sure if it's so relevant here

Comment (sai-bhargav) [reply gemfiles/rails_3.gemfile]: updated

File: .circleci/config.yml (hunk 1 above)
Comment (sai-bhargav) [reply .circleci/config.yml:8]: non-lts Rails 4 version and ruby 3 has compatibility issue resulting in issue with bundler version. Hence the build step fails to install bundler.

**Pull request merged**